pkg/cmd/release/download/download.go

```diff
@@ -174,6 +177,9 @@ func downloadRun(opts *DownloadOptions) error {
 			if len(opts.FilePatterns) > 0 && !matchAny(opts.FilePatterns, a.Name) {
 				continue
 			}
+			if runtime.GOOS == "windows" && isWindowsReservedFilename(a.Name) {
```

Comment:
Might consider making `isReservedFileName` a function from a `reserved_windows.go` and `reserved.go` to avoid this check at runtime?

Reply:
Yeah I thought about that. We have examples of both in the codebase. Do you have a strong opinion on which you prefer? I felt like this was a bit more clear from a reading code perspective and the runtime check cost did not seem terribly high. Happy to change it though.

Reply:
Add a comment that if we discover reserved file names on other systems we should change the pattern rather than add a conditional and I'm happy.

Reply:
Hello